File: gptproto-ts-sdk/scripts/local_gateway_contract.ts
import { GptprotoClient } from '../src';

const baseUrl = process.env.GPTPROTO_SDK_TEST_BASE_URL;
if (!baseUrl) throw new Error('GPTPROTO_SDK_TEST_BASE_URL is required');

const client = new GptprotoClient({ baseUrl });

try {
  await client.create('video', {
    model: 'sdk-test/not-a-real-model',
    prompt: 'SDK contract probe',
  });
  throw new Error('expected the local gateway to reject the probe model');
} catch (error) {
  const apiError = error as { status?: number; body?: { error?: { code?: number } } };
  if (apiError.status !== 400 || apiError.body?.error?.code !== 400) {
    throw error;
  }
  console.log('TypeScript gateway contract: ok');
}

File: gptproto-python-sdk/scripts/local_gateway_contract.py
"""Exercise the public Python SDK against a local gateway without submitting a channel task."""

import os

from gptproto import GptprotoClient
from gptproto.models import UnifiedVideoRequest


base_url = os.environ["GPTPROTO_SDK_TEST_BASE_URL"]
client = GptprotoClient(base_url=base_url)

try:
    client.create(
        "video",
        UnifiedVideoRequest(
            model="sdk-test/not-a-real-model",
            prompt="SDK contract probe",
        ),
    )
except RuntimeError as exc:
    if "code=400" not in str(exc):
        raise
    print("Python gateway contract: ok")
else:
    raise RuntimeError("expected the local gateway to reject the probe model")
